src/serverenvironment.cpp

```diff
@@ -254,40 +254,72 @@ void LBMManager::applyLBMs(ServerEnvironment *env, MapBlock *block,
 	FATAL_ERROR_IF(!m_query_mode,
 		"attempted to query on non fully set up LBMManager");
 
-	const v3s16 pos_of_block = block->getPosRelative();
-	v3s16 pos;
-	MapNode n;
-	content_t c;
-	auto it = getLBMsIntroducedAfter(stamp);
+	// Collect a list of all LBMs and associated positions
+	struct LBMToRun {
+		std::unordered_set<v3s16> p; // node positions
+		std::unordered_set<LoadingBlockModifierDef*> l;
```

Comment:
Why not vector?

Reply:
with the current structure insert is potentially called multiple times